File: packages/core/src/transport/MockTransport.ts
import type { Transport } from '../types.js';

export class MockTransport implements Transport {
  readonly sent: string[] = [];
  private handler: ((message: string) => void) | null = null;

  send(message: string): void {
    this.sent.push(message);
  }

  onMessage(handler: (message: string) => void): void {
    this.handler = handler;
  }

  simulateIncoming(message: string): void {
    this.handler?.(message);
  }

  dispose(): void {
    this.handler = null;
  }
}

File: android/core/src/main/kotlin/io/rynbridge/core/MockTransport.kt
package io.rynbridge.core

class MockTransport : Transport {

    private val lock = Any()
    private val _sent = mutableListOf<String>()
    private var messageHandler: ((String) -> Unit)? = null

    val sent: List<String>
        get() = synchronized(lock) { _sent.toList() }

    override fun send(message: String) {
        synchronized(lock) {
            _sent.add(message)
        }
    }

    override fun onMessage(handler: (String) -> Unit) {
        synchronized(lock) {
            messageHandler = handler
        }
    }

    override fun dispose() {
        synchronized(lock) {
            messageHandler = null
        }
    }

    fun simulateIncoming(message: String) {
        val handler = synchronized(lock) { messageHandler }
        handler?.invoke(message)
    }

    fun reset() {
        synchronized(lock) {
            _sent.clear()
        }
    }
}
